Firebase/Database/Login/FAuthTokenProvider.m

```diff
@@ -64,45 +67,52 @@ - (void) dealloc {
 
 @interface FIRFirebaseAuthTokenProvider : NSObject <FAuthTokenProvider>
 
-@property (nonatomic, strong) FIRApp *app;
+@property (nonatomic, strong) id<FIRAuthInterop> auth;
 /** Strong references to the auth listeners as they are only weak in FIRFirebaseApp */
 @property (nonatomic, strong) NSMutableArray *authListeners;
 
-- (instancetype) initWithFirebaseApp:(FIRApp *)app;
+- (instancetype) initWithAuth:(id<FIRAuthInterop>)auth;
 
 @end
 
 @implementation FIRFirebaseAuthTokenProvider
 
-- (instancetype) initWithFirebaseApp:(FIRApp *)app {
+- (instancetype)initWithAuth:(id<FIRAuthInterop>)auth {
     self = [super init];
     if (self != nil) {
-        self->_app = app;
+        self->_auth = auth;
         self->_authListeners = [NSMutableArray array];
     }
     return self;
 }
 
-- (void) fetchTokenForcingRefresh:(BOOL)forceRefresh withCallback:(fbt_void_nsstring_nserror)callback {
-    // TODO: Don't fetch token if there is no current user
-    [self.app getTokenForcingRefresh:forceRefresh withCallback:^(NSString * _Nullable token, NSError * _Nullable error) {
+- (void)fetchTokenForcingRefresh:(BOOL)forceRefresh
+                     withCallback:(fbt_void_nsstring_nserror)callback {
+  if (self.auth == nil) {
+    // Signal that Auth is not available by returning nil.
+    callback(nil, nil);
+  } else {
+    [self.auth getTokenForcingRefresh:forceRefresh
```

Comment:
Based on the `TODO` above, we should also check if the `getUserID` call returns nil and not call `getTokenForcingRefresh:withCallback`.

Reply:
Done.